src/styles/block/main.scss

```diff
@@ -0,0 +1,138 @@
+:root {
+  --main-columns-number: 1;
+}
+
+.main {
+  margin: 0;
+  font-family: Roboto, sans-serif;
+  display: grid;
+  grid-template-columns: repeat(var(--main-columns-number), 1fr);
+  grid-template-rows: repeat(2, 1fr);
+  padding: 50px 40px;
+  gap: 48px 46px;
+  justify-content: center;
+}
+
+@media (min-width: 488px) {
+  :root {
+    --main-columns-number: 2;
+  }
+}
+
+@media (min-width: 768px) {
+  :root {
+    --main-columns-number: 3;
+  }
+}
+
+@media (min-width: 1024px) {
+  :root {
+    --main-columns-number: 4;
+  }
+}
+
+.card {
+  width: 200px;
+  box-sizing: border-box;
+  padding-inline: 16px;
+
+  transition: transform $transition-time;
```

Comment:
The variable `$transition-time` is used here, but it's not defined in this file or in the provided context. Ensure that it is defined in a global scope (e.g., in `variables.scss`) to avoid any compilation errors.